### ISSUE TITLE
[Request]: Multi-column layout

### ISSUE THREAD

**Issue by cherti** [GeopJr/Tuba#118]
### Describe the request

I'd like to propose for consideration that Tuba gets the ability to choose multiple columns, similar to the Mastodon webinterface. It is very useful to have one column as the home feed, another one for notifications and another one for, for example, the home timeline or a specific timeline/followed hashtag/list. A multi-column layout would help with that very much, particularly when Tuba is used not as a slim client but in a wider window configuration. I don't know whether that applies to only some Fediverse implementations or all of them, but for some it would be extremely useful.

### Implementation Details

- [ ] This should be an option in settings.
- [ ] This should be only available to some fediverse backends. (Include which ones on the above field).
- [X] This is client-only (and shouldn't sync with the instance).
- [X] This follows the [GNOME HIG](https://developer.gnome.org/hig/).

**Comment by Nico7as** [comment]
I'd like that very much too !!

I looked for Linux apps, I found Sengi, as mentioned by nekohayo, and Fedistar ( https://github.com/h3poteto/fedistar ).
They both are Electron Apps, and not available on Flathub :-/

Is it possible with GTK and Libadwaita?

**Comment by GeopJr** [comment]
There are bigger fish to fry at the moment.

- Is it planned?
Not sure right now. I'm open to it but it clearly sacrifices being adaptive for all window sizes, as it requires a big minimum width.

- What are the bigger fish?
#500 and everything linked there and in the linked ones. The way widgets are not recycled means they stay in memory until cleared. With a multi-column layout, that means that issues * N timelines.

- Will it become a Mona replacement?
Isn't Mona a Mac app? I don't have access to a Mac so I won't really be able to debug issues there (even though it should be buildable and work according to #958). I don't know if Tuba would be a good replacement there

**Comment by maikelthedev** [comment]
> ### Describe the request
> 
> I'd like to propose for consideration that Tuba gets the ability to choose multiple columns, similar to the Mastodon webinterface. It is very useful to have one column as the home feed, another one for notifications and another one for, for example, the home timeline or a specific timeline/followed hashtag/list. A multi-column layout would help with that very much, particularly when Tuba is used not as a slim client but in a wider window configuration. I don't know whether that applies to only some Fediverse implementations or all of them, but for some it would be extremely useful.
> ### Implementation Details
> 
>     * [ ]  This should be an option in settings.[ ]  This should be only available to some fediverse backends. (Include which ones on the above field).[x]  This is client-only (and shouldn't sync with the instance).[x]  This follows the [GNOME HIG](https://developer.gnome.org/hig/).

this is th eone thing that would make me abandon entirely Mastodon vanilla web advanced interface. Also i made some adjustments to it with a firefox extension that I could bring to Tuba if it had the multi-column format. Check [this](https://codeberg.org/maikelthedev/mmi#screenshots) or see below

<img width="1421" height="767" alt="Image" src="https://github.com/user-attachments/assets/a306f671-64b7-4651-9578-a8f354f18717" />

**Comment by maikelthedev** [comment]
> On mobile

mobile? Isn't this a Linux app?

**Comment by GeopJr** [comment]
Hey all, I'd like to emphasize the "What are the bigger fish?" of my first comment. Implementing this is somewhat easy and straight-forward (at first glance at least) but having it as an option without dealing with the recycler situation, is just begging for more and more "memory leak" and "performance issues" tickets (since it will highlight them) and posts that I have to reply to with the same old message explaining the whole (frustrating) situation.

Best I could do in the meantime is maybe have this as a hidden option instead of advertising as something 100% supported. As for mobile / narrow window size, we could do maybe a horizontally scrolling view? This would also allow for an infinite amount of columns without requiring a minimum window size.

**Comment by maikelthedev** [comment]
> Best I could do in the meantime is maybe have this as a hidden option instead of advertising as something 100% supported

yes please, if it creates issues once is there we can fork and help you fix them.
